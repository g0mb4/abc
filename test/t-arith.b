main() {
    extrn putchar, printn;
    auto a, b, c;

    a = 2;
    b = 3;

    /* 5 */
    c = b + a;
    printn(c, 10);
    putchar('*n');

    /* 1 */
    c = b - a;
    printn(c, 10);
    putchar('*n');

    /* 6 */
    c = b * a;
    printn(c, 10);
    putchar('*n');

    /* 3 */
    c = 6 / a;
    printn(c, 10);
    putchar('*n');
    
    /* 2 */
    c = 6 % 4;
    printn(c, 10);
    putchar('*n');

    /* 7 */
    printn(1+2*3, 10);
    putchar('*n');

    /* 9 */
    printn((1+2)*3, 10);
    putchar('*n');

    /* 9 */
    printn(3*(1+2), 10);
    putchar('*n');
}
