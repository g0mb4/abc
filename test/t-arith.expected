5
1
6
3
2
7
9
9
